lgc/patch/PatchBufferOp.cpp

```diff
@@ -420,12 +420,39 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
   } else if (callName.startswith(lgcName::LateBufferLength)) {
     Value *const pointer = getPointerOperandAsInst(callInst.getArgOperand(0));
 
-    // Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
-    Value *const bufferLength = m_builder->CreateExtractElement(m_replacementMap[pointer].first, 2);
+    // Compute the length of the variable sized array
+    // 1. Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
+    Value *const bufferDesc = m_replacementMap[pointer].first;
+    Value *numRecords = m_builder->CreateExtractElement(bufferDesc, 2);
+
+    Value *offset = callInst.getArgOperand(1);
+    if (cast<ConstantInt>(offset)->getZExtValue() > 0) {
+      // 2. Get the number records of the variable sized array by substracting the byte offset
+      numRecords = m_builder->CreateSub(numRecords, offset);
+    }
+
+    // 3. Extract STRIDE (SQ_BUF_RSRC_WORD1 [29:16]) and divide into numRecords
```

Comment:
I think you should leave the divide where it was before, in SPIRVReader. I think putting it here and relying on the buffer descriptor stride being the same as the array stride is probably wrong anyway.

Reply:
Done.